File: SocialNetwork.WebApp/src/types/post.ts
import { MediaType } from "../enums/media";
import { PrivacyType } from "../enums/privacy";
import { GroupResource } from "./group";
import { TagResource } from "./tag";
import { UserResource } from "./user";

export type PostResource = {
    id: string;
    content: string;
    background: string;
    isGroupPost: boolean;
    privacy: PrivacyType;
    user: UserResource;
    createdAt: Date;
    medias: PostMediaResource[];
    comments: number;
    shares: number;
    reactions: number;
    postType: string;
    originalPostId: string;
    sharePost: PostResource;
    originalPost: PostResource;
    sharePostId: string;
    group: GroupResource
    tags: TagResource[];
    isSaved: boolean;
}

export type PostMediaResource = {
    id: string;
    mediaUrl: string;
    mediaType: MediaType
}

File: SocialNetwork.Domain/Entity/PostInfo/Post.cs
using SocialNetwork.Domain.Abstractions;
using SocialNetwork.Domain.Entity.GroupInfo;
using SocialNetwork.Domain.Entity.System;

namespace SocialNetwork.Domain.Entity.PostInfo
{
    public class Post : BaseAuditableEntity
    {
        public string Content { get; set; }
        public string? Background { get; set; }
        public string Privacy { get; set; }
        public string PostType { get; set; }
        public bool IsGroupPost { get; set; }
        public bool IsApproved { get; set; }

        public Guid? GroupId { get; set; }
        public Group? Group { get; set; }

        public string UserId { get; set; }
        public User User { get; set; }

        public Guid? SharePostId { get; set; }
        public Post SharePost { get; set; }

        public Guid? OriginalPostId { get; set; }
        public Post OriginalPost { get; set; }

        public ICollection<Post> Shares { get; set; } = new List<Post>();
        public ICollection<Comment> Comments { get; set; }
        public ICollection<Tag> Tags { get; set; }
        public ICollection<Reaction> Reactions { get; set; }
        public ICollection<PostMedia> Medias { get; set; }
    }
}
